test/groovy/DebugReportArchiveTest.groovy

```diff
@@ -0,0 +1,53 @@
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import util.*
+
+import static org.hamcrest.Matchers.containsString
+import static org.junit.Assert.assertThat
+
+class DebugReportArchiveTest extends BasePiperTest {
+
+    private JenkinsLoggingRule loggingRule = new JenkinsLoggingRule(this)
+    private JenkinsStepRule stepRule = new JenkinsStepRule(this)
+    private JenkinsWriteFileRule writeFileRule = new JenkinsWriteFileRule(this)
+
+    @Rule
+    public RuleChain ruleChain = Rules
+        .getCommonRules(this)
+        .around(new JenkinsReadYamlRule(this))
+        .around(loggingRule)
+        .around(writeFileRule)
+        .around(stepRule)
+
+    @Before
+    void init() {
+        helper.registerAllowedMethod("libraryResource", [String.class], { path ->
+
+            File resource = new File(new File('resources'), path)
+            if (resource.exists()) {
+                return resource.getText()
+            }
+
+            return ''
+        })
+    }
+
+    @Test
+    void testDebugReportArchive() {
+        stepRule.step.debugReportArchive(
+            script: nullScript,
+            juStabUtils: utils,
+            stageName: 'test',
+            printToConsole: true
+        )
+
+        String debugReportSnippet = 'The debug log is generated with each build and should be included in every support request'
+
+        assertThat(loggingRule.log, containsString('Successfully archived debug report'))
+        assertThat(loggingRule.log, containsString(debugReportSnippet))
+
+        assertThat(writeFileRule.files.values()[0] as String, containsString(debugReportSnippet))
```

Comment:
Can we access the file over the name instead of using an index?

Reply:
The filename is currently generated based on a time-stamp and fields of DebugReport. That isn't so nice in the first place... will try to fix.

Reply:
done